keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/resource/ImageHandler.kt

```diff
@@ -44,48 +46,57 @@ class ImageHandler(
   override fun generateName(spec: ImageSpec): ResourceName =
     ResourceName("bakery:image:${spec.artifactName}")
 
-  override fun desired(resource: Resource<ImageSpec>): Image =
-    with(resource.spec) {
+  override fun resolve(resource: Resource<ImageSpec>): ResolvedResource<Image> =
+    with(resource) {
+      val artifact = DeliveryArtifact(spec.artifactName, DEB)
       val latestVersion = artifactRepository
-        .versions(DeliveryArtifact(artifactName, DEB))
-        .first()
-      Image(
-        baseAmiVersion = baseImageCache.getBaseImage(baseOs, baseLabel),
-        // TODO: there must be a canonical way of composing this
-        appVersion = latestVersion,
-        regions = regions
-      )
+        .versions(artifact)
+        .firstOrNull() ?: throw NoKnownArtifactVersions(artifact)
+      runBlocking {
+        ResolvedResource(
+          desired = desired(resource.spec, latestVersion),
+          current = current(resource.spec, latestVersion)
+        )
+      }
```

Comment:
Now we resolve the AMI for the specific artifact version so really what we're checking is a) does the AMI exist at all? and b) is it using the current baseAMI?